packages/types/src/primitive/StorageKey.ts

```diff
@@ -70,6 +70,14 @@ export default class StorageKey extends Bytes {
 
       assert(isFunction(fn), 'Expected function input for key construction');
 
+      if (Array.isArray(arg[0])) {
```

Comment:
Can we just add a comment here to indicate why we are doing this? (2 months from now this will look like black magic without background/info)

Reply:
Thanks for the review. Yeah, I will add some commands here.